AgContext.yaml

```diff
@@ -0,0 +1,321 @@
+type: object
+required:
+  - id
+  - crop_type
+  - crop_stage
+  - soil_colour
+  - surface_cover
+  - surface_coverage
+  - weather_description
+  - location_lat
+  - location_long
+  - location_datum
+  - camera_model
+  - lens_model
+  - camera_height
+  - camera_angle
+  - camera_fov
+  - photography_description
+  - cropped_to_plant
+
+properties:
+
+  id:
+    type:
+      number
+    description: |-
+        A number identifying the AgContext.
+
+  crop_type:
+    type:
+      string
+    description: |-
+          General crop type.
+          A string describing the general cropping scenario (e.g. "root vegetable").
+
+  grains_descriptive_text:
+    type:
+      string
+    enum:
+      - emergence
+      - seedling
+      - tillering
+      - stem_elongation
+      - booting
+      - ear_emergence
+      - flowering
+      - milky_dough
+      - dough
+      - ripening
+      - na
+    description: |-
+        Grains descriptive text.
+        One of ten possible strings describing the crop developmental stage.
+        If this AgContext is not in a cropping environment, use value "na".
+
+  bbch_descriptive_text:
+    type:
+      string
+    enum:
+      - germination
+      - sprouting
+      - bud development
+      - leaf development
+      - formation of side shoots
+      - tillering
+      - stem elongation
+      - rosette growth
+      - shoot development
+      - development of harvestable vegetative parts
+      - bolting
+      - inflorescence emergence
+      - heading
+      - flowering
+      - development of fruit
+      - ripening or maturity of fruit and seed
+      - senescence
+      - beginning of dormancy
+      - na
+    description: |-
+        BBCG descriptive text.
+        One of several possible strings describing the stage of the crop, chosen from a list of possible terms used by the BBCH.
+        If this AgContext is not in a cropping environment, use value "na".
+
+  bbch_code:
```

Comment:
This is a bit silly looking; would something like a regex be possible here?

Reply:
A regex is certainly possible. https://json-schema.org/understanding-json-schema/reference/regular_expressions.html `pattern: "^gs[0-9][0-9]$|^na$`. Although this includes gs00 which is currently not permitted here. (We can exclude that from the pattern if necessary.)